packages/shared/src/baduk/badukWithAbstractBoard.ts

```diff
@@ -0,0 +1,269 @@
+import { AbstractGame } from "../abstract_game";
+import type { MovesType } from "../abstract_game";
+import type { IBadukBoard } from "../Interfaces/IBadukBoard";
+import { BadukBoardAbstract } from "./board/BadukBoardAbstract";
+import type { Intersection } from "./board/intersection";
+
+export enum Color {
+    EMPTY = 0,
+    BLACK = 1,
+    WHITE = 2,
+}
+
+export enum BoardPattern {
+    Uknown = 0,
+    Rectangular = 1,
+    Polygonal = 2,
+}
+
+export interface BadukWithAbstractBoardConfig {
+    width: number;
```

Comment:
I tried to build this so it can handle both rectangular and polygonal boards, hence the config has width and height. The polygonal board then takes the minimum of the both as size (which is arbitrary, but works for now).

Reply:
Does min(width, height) just mean the number of tiles then? No matter what combo of width and height I enter, it seems like the same shape is output:

<img width="400" alt="Screen Shot 2022-09-04 at 9 37 08 AM" src="https://user-images.githubusercontent.com/25233703/188323934-25412578-9299-46b4-bdfb-ba063f27208e.png">

Reply:
I think that the game logic in badukWithAbstractBoard is not yet correctly connected to the view in the vue-client. There the component defined in BadukBoardAbstract.vue creates its own board. I'm going to connect the two when I have more time (next weekend), but for now you can try different board sizes by changing the value of the variable boardsize in BadukBoardAbstract.vue